drivers/input/input_gt911.c

```diff
@@ -244,13 +254,30 @@ static int gt911_init(const struct device *dev)
 	uint32_t reg_id = 0;
 	uint16_t reg_addr = DEVICE_ID;
 
-	r = i2c_write_read_dt(&config->bus, &reg_addr, sizeof(reg_addr),
+	r = i2c_write_read_dt(&data->bus, &reg_addr, sizeof(reg_addr),
 							&reg_id, sizeof(reg_id));
 	if (r < 0) {
-		return r;
+		/* Some boards may route the interrupt GPIO as an input
+		 * only. In this case, the address selection code above
```

Comment:
what you mean by "address selection code"?

Reply:
The GT911 uses either 0x5D or 0x14 as the I2C address, based on the state of the INT pin during reset:

0x14:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/5cd4aa7a-ad8c-4d50-9290-59a5b61413a9)
0x5D:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/8fbdf895-cbd6-4865-99de-04f6d71e8de2)

The code here attempts to select 0x5D, but *some boards* (the RT595 EVK) pass the INT signal through a level shifter, and only support receiving inputs on this pin. This would also be an issue for a board that did not route the INT signal. The workaround here is basically to try the other address the controller might use if 0x5D fails, since the INT line is floating during this reset period on the RT595 EVK.

Reply:
Ok so it's a workaround for boards that have no defined state of the strap at reset time, I'm still a bit confused by the use case, the INT pin has no strapping either internally (the datasheet does not mention it) or externally before the level shifter? Weird design choice, but I guess fine let's support this.

Anyway, I still find the "above" wording somewhat confusing, is there a code above this code that does something address selection?

Reply:
Oh I see what you mean now, the reset sequence thing! Ok ignore this comment. :-)

Reply:
> Weird design choice, but I guess fine let's support this.

I agree- I can't find anything indicating that the INT pin would have a strapped state on the RT595 EVK. The INT pin is routed through a level shifter: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/494f0865-556c-47ba-83a0-a59ed17fcfe7)

In the NXP MCUX SDK the GT911 driver also does this probing step, to resolve the same issue. I only see it reproduced on one of our display boards (the RK055HDMIPI4MA0), and to induce the error I usually need to toggle power to the EVK. A warm reset won't usually cause issues. Regardless, I think we have to work around this hardware issue in software

Reply:
> Oh I see what you mean now, the reset sequence thing! Ok ignore this comment. :-)

I'll add a bit more specifics to the comment in this PR- if it confused you, it will confuse someone else in the future.

Reply:
> ^^ this is probably not very nice to the GPIO if on the other side there's a directional level translator also trying to drive the pin as hard as it can. Should that sequence be optional?

Yeah, I expect it isn't. One way we could make this optional would be to skip the INT pin setting code if `config->alt_addr` was set, and just fall though to address probing. Thoughts on that approach?